### PR TITLE
Initial NCoV-Tools Implementation

### PR DESCRIPTION
**Pull request by DarianHole**
Starting point for implementation of [ncov-tools](https://github.com/jts/ncov-tools)

### Points to think about:

- output pdf files have plots that have the sample name as `<prefix>_barcode**.fastq` while I've set the output for irida upload to have the correct sample names with a python script

    - May be able to rename the files before running the tool in the `run` directory

        - Again with a python script or with some sort of additional bash commands although I don't know a good simple bash method to rename files based on an input csv file

    - May be able to modify the initial parts of the script to rename then, but that may end up with troubleshooting a lot of pipeline issues or it may be really easy

        - Still would need a good way to rename the files from a csv file

- Potentially better ways to put the input files in

    - Currently uses `.toList().flatten().toList` to link all output files of nanopolish results

    - Could find a good way to use the singular output channels for the input `*.consensus.fasta` and `*.mapped.bam` files although they are in tuples and I was having trouble using them as an input and getting them all to be in the same spot at once

- Could also just set up a new workflow using the base pipeline that solely suits the NML needs allowing easier updates to connor-lab pipeline and potentially a bit more modification to certain inputs, outputs, and steps

### CODE REVIEW

File: environments/ncovtools.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - bioconda
+  - defaults
+dependencies:
+  - snakemake
+  - mafft
+  - augur
+  - pysam
+  - r-base=3.6.1
+  - iqtree<2
+  - bioconductor-ggtree=2.0.0
```
Comment (Takadonet) [inline environments/ncovtools.yml:12]: can you add r-argparse ? Just was added yesterday

File: modules/nml.nf [hunk 2]
```diff
@@ -18,3 +18,30 @@ process generateIridaReport {
...
+    file("*.pdf")
+
+    script:
+    """
+    git clone https://github.com/jts/ncov-tools.git
+    mv ${config} ${reference} ${amplicon} ./ncov-tools
+    mkdir ./ncov-tools/run
+    mv ${params.prefix}* ./ncov-tools/run
+    cd ncov-tools
+    snakemake -s qc/Snakefile all_qc_sequencing --cores 6
```
Comment (Takadonet) [inline modules/nml.nf:44]: Can you get value for --cores # from https://github.com/phac-nml/ncov2019-artic-nf/blob/master/conf/coguk/nml.config#L6

Perhaps a new label so we can adjust as needed without changing nextflow code.

Comment (DarianHole) [reply modules/nml.nf:44]: Do you want me to look into a way to grab the value from the config file or just create a new label and match the cores to that?

**Comment by Takadonet** [comment]
New label and match

File: environments/ncovtools.yml (hunk 1 above)
Comment (DarianHole) [reply environments/ncovtools.yml:13]: Added

File: modules/nml.nf (hunk 2 above)
Comment (DarianHole) [reply modules/nml.nf]: Believe I addressed this in the way you wanted

**Comment by DarianHole** [comment]
Merging in. Next PR will address issues that I have noted in this PR

**Pull request merged**